File: src/main/java/org/taonaw/reservation/domain/model/rentalequipments/RentalEquipment.java
package org.taonaw.reservation.domain.model.rentalequipments;

import lombok.NonNull;

public class RentalEquipment {
    private final RentalEquipmentId equipmentId;
    private final RentalEquipmentStock rentalEquipmentStock;

    private RentalEquipment(
            @NonNull RentalEquipmentId equipmentId,
            @NonNull RentalEquipmentStock rentalEquipmentStock) {
        this.equipmentId = equipmentId;
        this.rentalEquipmentStock = rentalEquipmentStock;
    }
}

File: src/backend/reservation/src/main/kotlin/org/taonaw/reservation/domain/model/reservation/RentalEquipment.kt
package org.taonaw.reservation.domain.model.reservation

import org.taonaw.reservation.domain.model.reservationPolicy.MaxRentalEquipmentQuantity
import org.taonaw.reservation.domain.model.shared.EquipmentId

data class RentalEquipment(
        val equipmentId: EquipmentId,
        val quantity: RentalEquipmentQuantity) {

    fun satisfy(maxRentalEquipmentQuantity: MaxRentalEquipmentQuantity): Boolean {
        return quantity.satisfy(maxRentalEquipmentQuantity)
    }
}